src/targets/gpu/hip.cpp

```diff
@@ -187,10 +187,28 @@ argument register_on_gpu(const argument& arg)
     return {s, [p, a = std::move(arg_shared)]() mutable { return get_device_ptr(p.get()); }};
 }
 
+// argument to_gpu(const argument& arg, bool host)
+//{
+//    auto p = write_to_gpu(arg.data(), arg.get_shape().bytes(), host);
+//    return {arg.get_shape(), p};
+//}
+
 argument to_gpu(const argument& arg, bool host)
 {
-    auto p = write_to_gpu(arg.data(), arg.get_shape().bytes(), host);
-    return {arg.get_shape(), p};
+    argument result;
```

Comment:
This can be a separate fix : 
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/1465/files

Reply:
You can add unit-tests with that.

Reply:
Added a unit test. The change is required to run the driver. I could separate it, but it would likely lead to longer review time.